### PR TITLE
Refactoring of pb_list() and other changes

### LINKED ISSUE #60: pb_list() should not offer to create a new release if no releases are found
I think pb_info() does too much right now and would be less surprising if it only offered to create a new release in specific situations, e.g. when trying to upload something. It doesn't make sense if I'm querying whether there is release data (e.g. `pb_list()`) 

```
R> piggyback::pb_list("tanho63/tanho63")
No releases found would you like to create one? (yes/no/cancel) no
```

I can make a PR to add a parameter to pb_info so that it can be configured to prompt you (and default to offer creation since that's the current default and would maintain backcompat)

### LINKED ISSUE #61: pb_list() will soon fail if multiple tags are specified
If I'm querying a repository for files attached to multiple tags, it currently "works" in that it returns the expected dataframe but has the following warnings:
```
R> x <-   piggyback::pb_list(repo = "nflverse/nflverse-data", tag = c("pbp","player_stats"))
Warning messages:                                
1: In if (tag == "latest") { :
  the condition has length > 1 and only the first element will be used
2: In if (tag %in% info$tag) { :
  the condition has length > 1 and only the first element will be used
```

These warnings will soon be an error as of R 4.2+ https://twitter.com/henrikbengtsson/status/1496597313510273026?s=20 - so properly supporting vectorized tags would be useful! 

I think this again refers to `pb_info()` related issues.

### LINKED ISSUE #63: memoise should happen onLoad and not at build time
as per suggestions here: https://github.com/r-lib/memoise/issues/76

### LINKED ISSUE #64: simplify various CLI messaging package dependencies
I think the latest package by Gabor is {cli}, which is imported by gh already - so I think it would be useful to remove dependencies on clisymbols and crayon by consistently using cli instead

### PR DESCRIPTION
**Pull request by tanho63**
- pb_releases() created and exported, returns a list of releases available in repo
- pb_info rewritten to handle multiple and specific tags, resolves #61 
- pb_info now checks for specified tags first and only returns assets for those tags if specified
- pb_info no longer asks about creating new releases, nor does it warn about missing releases - resolves #60
- pb_upload asks about creating a new release if no release found - correct spot to ask about releases in #60
- memoise moved to .onLoad() per best practice, resolves #63
- updating CLI messaging to use {cli} instead of {clisymbols} and {crayon} to reduce dependencies (gh imports cli and that seems to be latest best practice) resolves #64 

TODO: 
- [x] make all tests pass
- [x] update news file
- [x] audit rest of code for print messaging improvements

### CODE REVIEW

**Comment by tanho63** [comment]
Okay, so this PR got a little bit away from me and does quite a lot, but I think it takes care of a lot of the outstanding issues I noticed when I was working through it. 

Please let me know if you see any issues or have any questions!

File: R/zzz.R [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.onLoad <- function(libname,pkgname){
+  cache_duration <- as.numeric(Sys.getenv("piggyback_cache_duration", unset = "600"))
+  assign(x = "pb_info",
+         value = memoise::memoise(pb_info, ~ memoise::timeout(cache_duration)),
+         envir = parent.env(environment()))
+  assign(x = "pb_releases",
+         value = memoise::memoise(pb_releases, ~ memoise::timeout(cache_duration)),
+         envir = parent.env(environment()))
```
Comment (cboettig) [inline R/zzz.R:8]: Nice! guessing this syntax should avoid CRAN mistakenly complaining that `<<-` assigns to the global env

**Review by cboettig — APPROVED** [review]
All looks good to me. Thanks for the excellent work here!  Will cut a new CRAN release to ship these changes shortly.

File: R/zzz.R (hunk 1 above)
Comment (tanho63) [reply R/zzz.R:8]: Yep exactly!

**Comment by tanho63** [comment]
Awesome! Let me know if there's anything I can help with re: CRAN release

**Pull request merged**